### PR TITLE
Add opt-in colors to the CLI

### PR DESCRIPTION
**Pull request by brianstrauch**
Release Notes
-------------
New Features
- Optionally enable text highlighting with `confluent configuration update enable_colors true`

Checklist
---------
1. [CRUCIAL] Is the change for features that are already live in prod?
   * yes: ok

What
----
Styling for errors (red), code snippets (red with gray background), and links (blue with underline). Inference for dark and light terminals. Colors are not displayed by default can be enabled with `confluent configuration update enable_colors true`.

I considered adding highlighting for resources, i.e. "lkc-123456", but there are too many false positives at the moment (other text in quotes).

Screenshot 2023-08-25 at 5 10 44 PM

Screenshot 2023-08-25 at 5 11 09 PM

Screenshot 2023-08-25 at 6 39 56 PM

Test & Review
-------------
Manual verification

### CODE REVIEW

File: internal/audit-log/command_config_migrate.go
Comment (sgagniere) [inline internal/audit-log/command_config_migrate.go:83]: It looks like the `output` package is still around in this PR and being used in a few places; what's the general guideline for when to use `output` and when to use `color`? Or could we fully replace output with color?

File: internal/admin/command_payment_describe.go
Comment (sgagniere) [inline internal/admin/command_payment_describe.go:34]: Is there any good way to hide this parameter? Like, what if we make a color or printer struct holding this bool, make the print functions into methods for that struct, and add it to the command structs, so that we could write something like `c.Color.Printf("Organization is...")`.

**Review by sgagniere — COMMENTED** [review]
Nice; this will be a good quality of life upgrade. But I have some questions:

File: pkg/cmd/errors.go
Comment (MuweiHe) [inline pkg/cmd/errors.go:10]: This might be put together in pkg errors?

File: internal/audit-log/command_config_migrate.go
Comment (MuweiHe) [reply internal/audit-log/command_config_migrate.go:83]: +1

Comment (brianstrauch) [reply internal/audit-log/command_config_migrate.go]: Good idea, updated to fully replace color with output to avoid future confusion!

File: internal/admin/command_payment_describe.go
Comment (brianstrauch) [reply internal/admin/command_payment_describe.go]: Not all print statements have access to `c` at the moment, but I think they all should be rewritten to... I think after we do that then making this change would be a good idea. In the meantime, let's just have a single set of functions, `output.Print*()` as we discussed in the comment above

File: pkg/cmd/errors.go
Comment (brianstrauch) [reply pkg/cmd/errors.go:10]: I put this function in this package since it (and other functions in this package) mirrors a function inside the `cobra` package.

File: internal/schema-registry/schema_utilities.go
Comment (sgagniere) [inline internal/schema-registry/schema_utilities.go:67]: I'm a bit unclear still on when we use `c.Config.EnableColor` and when we hardcode `false`.

**Review by sgagniere — COMMENTED** [review]
A few more things:

File: internal/schema-registry/schema_utilities.go
Comment (brianstrauch) [reply internal/schema-registry/schema_utilities.go:67]: The majority of the time, `c` is not available and making it available would result in a huge rewrite. My plan is to gradually rewrite these one by one. For the time being, we use `false` which corresponds to no highlighting, even when colors are enabled.

Comment (sgagniere) [reply internal/schema-registry/schema_utilities.go:67]: Oh woops, you addressed this last time. Gotcha. Lgtm after the other comment.

File: internal/command.go
Comment (brianstrauch) [reply internal/command.go:69]: No, this is because the by default, errors are printed by cobra in black and white. I was hoping I could disable this and print my own (colored) errors manually. Unfortunately it's proving to be more difficult than I thought...

**Review by sgagniere — APPROVED** [review]

**Pull request merged**
